//python -m grpc_tools.protoc -I./proto --python_out=. --grpc_python_out=. ./proto/tracker.proto
syntax = "proto3";

service Swarm{
    rpc CreateSwarm(SwarmNode) returns (Status){}  
    rpc RequestSwarm(SwarmData) returns (Status){} //cambiar por string
    rpc AddToSwarm(SwarmNode) returns (Status){}
}
//Para anunciarce
message SwarmNode{
    string fileName = 1;
    string seederIP = 2;
    int32 seederPort = 3;
    string id = 4;
    //Agregar un id
}
message Status{
    string details = 1;
}

//Para obtener lista
message SwarmData{
    string fileName = 1;
    string leecherIP = 2;
    string leecherPort = 3;
    string id = 4;
}

//Para a√±adir al Swarm
